src/core/FeedsManager.cpp

```diff
@@ -66,6 +66,16 @@ void Feed::markEntryAsRead(const QString &identifier)
 	}
 }
 
+void Feed::markAllEntriesAsRead()
+{
+	for (int i = 0; i < m_entries.count(); ++i)
+	{
+		m_entries[i].lastReadTime = QDateTime::currentDateTimeUtc();
```

Comment:
Store current date and time in a `const` variable before loop (for example `lastReadTime`) instead of calling `QDateTime::currentDateTimeUtc()` all the time.
Also, there should be a check to ignore entries that were already marked as read.